src/main/java/edu/wpi/grip/core/Socket.java

```diff
@@ -36,6 +39,7 @@ public Socket(EventBus eventBus, SocketHint<T> socketHint, T value) {
         this.eventBus = eventBus;
         this.socketHint = socketHint;
         this.value = value;
+        this.direction = Direction.INPUT;
```

Comment:
I don't like that this assumes a socket is automatically an input. I think forcing you to explicitly define the direction is better practice.